### PR TITLE
adjust sso mapping fields to prevent select disabled

### LINKED ISSUE #2218: Bug interface édition SSO
quand on coche "A un SSO?" et qu'on ajoute du mapping, le select est disabled

### PR DESCRIPTION
**Pull request by pabois**
close #2218

### CODE REVIEW

File: app/views/server/universities/_sso_mapping.html.erb [hunk 1]
```diff
@@ -84,6 +84,27 @@ end
         fields: <%= object.sso_mapping.blank? ? '[]' : object.sso_mapping.to_json.html_safe %>,
         keys: <%= mapping_keys.map { |key| [key, User.human_attribute_name(key)] }.to_h.to_json.html_safe %>
       }
+    },
+    methods: {
+      addField: function () {
+        this.fields.push({sso_key: 'key', internal_key: 'email', roles: {}});
+        this.unDisableSelects();
```
Comment (arnaudlevy) [inline app/views/server/universities/_sso_mapping.html.erb:91]:
```suggestion
        this.enableSelects();      
```

File: app/views/server/universities/_sso_mapping.html.erb [hunk 2]
```diff
@@ -84,6 +84,27 @@ end
         fields: <%= object.sso_mapping.blank? ? '[]' : object.sso_mapping.to_json.html_safe %>,
         keys: <%= mapping_keys.map { |key| [key, User.human_attribute_name(key)] }.to_h.to_json.html_safe %>
       }
+    },
+    methods: {
+      addField: function () {
+        this.fields.push({sso_key: 'key', internal_key: 'email', roles: {}});
+        this.unDisableSelects();      
+      },
+      unDisableSelects: function () {
```
Comment (arnaudlevy) [inline app/views/server/universities/_sso_mapping.html.erb:93]:
```suggestion
      enableSelects: function () {
```

File: app/views/server/universities/_sso_mapping.html.erb [hunk 3]
```diff
@@ -84,6 +84,27 @@ end
...
+          selectFields = document.getElementById("app").getElementsByClassName("form-select");
+          for (i = 0; i < selectFields.length; i += 1) {
+              target = selectFields[i];
+              target.removeAttribute('disabled');
+          }
+        }, 100);
+      }
+    },
+    mounted() {
+      this.unDisableSelects();
```
Comment (arnaudlevy) [inline app/views/server/universities/_sso_mapping.html.erb:107]:
```suggestion
      this.enableSelects();
```

**Review by arnaudlevy — COMMENTED** [review]
Double négation

**Review by SebouChu — APPROVED** [review]

**Pull request merged**